auth-server/src/services/auth.ts

```diff
@@ -323,6 +338,37 @@ export default class AuthService {
 		return this.getSignedToken(user, addresses);
 	}
 
+	public async ChangeNotificationPrefrence(token: string, post_participated: boolean, post_created: boolean, new_proposal: boolean, own_proposal: boolean) {
+		const userId = await getUserIdFromJWT(token, jwtPublicKey);
+		const user = await getUserFromUserId(userId);
+		let notification = await Notification
+			.query()
+			.where('user_id', user.id)
+			.first();
+
+		if (!notification) {
+			notification = await Notification
+				.query()
+				.allowInsert('[user_id, post_participated, post_created, new_proposal, own_proposal]')
+				.insert({
+					user_id: user.id,
+					...NOTIFICATION_DEFAULTS
+				});
+		}
+
+		const update = {
+			post_participated: post_participated === undefined ? notification.post_participated: post_participated,
```

Comment:
A little more concise:
```suggestion
			post_participated: post_participated ? notification.post_participated: post_participated,
```

If that works, we should use that for for each field.

Reply:
this will not work as post_participated could be false. I am specifically checking for only undefined here